src/Swarm/Game/Step.hs

```diff
@@ -631,10 +639,8 @@ stepCESK cesk = case cesk of
     case wc of
       Left exn -> return $ Up exn s k
       Right () -> do
-        forM_ rf $ \case
-          AddEntity c e -> robotInventory %= E.insertCount c e
-          LearnEntity e -> robotInventory %= E.insertCount 0 e
-        needsRedraw .= True
+        applyRobotUpdates rf
+        flagRedraw
```

Comment:
Can we abstract this even further into a single function called both here and in `finishCookingRecipe` when `remTime <= 0`?  As it stands, this seems ripe for a subtle bug where we decide to change the behavior of recipes somehow, and someone changes this code but forgets/doesn't realize that they also have to change the code in `finishCookingRecipe` to match, resulting in inconsistent behavior between recipes with and without `time: 0`.